python/kserve/kserve/utils/utils.py

```diff
@@ -156,7 +197,8 @@ def get_predict_response(payload: Union[Dict, InferRequest], result: Union[List,
                     name=col,
                     shape=list(result[col].shape),
                     datatype=from_np_dtype(result[col].dtype),
-                    data=result[col].tolist()
+                    data=[bytes(val, 'utf-8') if isinstance(val, str)
```

Comment:
This is gRPC specific which may affect REST, the better place might be to_grpc 
https://github.com/kserve/kserve/blob/master/python/kserve/kserve/protocol/infer_type.py#L280